dynamic-demo-plugin/src/utils/telemetry.ts

```diff
@@ -72,19 +72,27 @@ export const eventListener = async (eventType: string, properties?: any) => {
     },
   };
   switch (eventType) {
-    case 'identify':
+    case 'identify': {
       const { user, ...otherProperties } = properties;
-      const id = user.metadata.uid || `${location.host}-${user.metadata.name}`;
+      // With 4.15+ we can use the user object directly, but on older releases (<4.15)
+      // we need to extract the user object from the metadata.
+      // All properties are defined in the UserInfo interface and marked as optional.
+      const uid = user?.uid || user?.metadata?.uid;
```

Comment:
down the road we should be having helpers for getting these so we dont need to care about the path